Mathlib/Topology/ContinuousMap/StarOrdered.lean

```diff
@@ -3,63 +3,65 @@ Copyright (c) 2024 Jireh Loreaux. All rights reserved.
 Released under Apache 2.0 license as described in the file LICENSE.
 Authors: Jireh Loreaux
 -/
-import Mathlib.Analysis.Complex.Basic
-import Mathlib.Data.Real.StarOrdered
-import Mathlib.Topology.ContinuousMap.Lattice
 import Mathlib.Topology.ContinuousMap.ContinuousMapZero
+import Mathlib.Topology.ContinuousMap.Ordered
+import Mathlib.Algebra.Order.Star.Basic
+
+/-! # Continuous functions as a star-ordered ring
+
+The type class `ContinuousSqrt` gives a sufficient condition on `R` to make `C(α, R)`
+and `C(α, R)₀` into a `StarOrderedRing` for any topological space `α`, thereby providing a means
+by which we can ensure `C(α, R)` has this property. This condiiton is satisfied
+by `ℝ≥0`, `ℝ`, and `ℂ`, and the instances can be found in the file
+`Topology.ContinuousMap.ContinuousSqrt`.
+
+## Implementation notes
+
+Instead of asking for a well-behaved square root on `{x : R | 0 ≤ x}`, we ask for it on the set
+`{x : R × R | x.1 ≤ x.2}`. This is because we need this type class to work for `ℝ≥0` for the
```

Comment:
```suggestion
Instead of asking for a well-behaved square root on `{x : R | 0 ≤ x}` in the obvious way, we instead require that, for every `x y  : R` such that `x ≤ y`, there exist some `s` such that `x + s*s = y`. This is because we need this type class to work for `ℝ≥0` for the
```
I think this is much clearer, I have no idea what a well-behaved square root on `{x : R × R | x.1 ≤ x.2}` means without scrolling down to the actual definition.